gpytorch/models/exact_prediction_strategies.py

```diff
@@ -117,9 +117,14 @@ def get_fantasy_strategy(self, inputs, targets, full_inputs, full_targets, full_
         # Evaluate fant x train and fant x fant covariance matrices, leave train x train unevaluated.
         fant_fant_covar = full_covar[..., num_train:, num_train:]
         fant_mean = full_mean[..., num_train:]
-        mvn = self.likelihood(MultivariateNormal(fant_mean, fant_fant_covar), inputs)
-        fant_fant_covar = mvn.covariance_matrix
+        mvn = MultivariateNormal(fant_mean, fant_fant_covar)
+        self.likelihood.fantasize(**kwargs)
+        if "noise" in kwargs:
```

Comment:
If we make the kwarg `observation_noise` in both places, we can just pass `**kwargs` along to the likelihood call and avoid this if/else branch.

Reply:
Yeah so the reason I chose `noise` instead of `observation_noise` in fantasize is to have some distinction there - what's passed into fantasize will become part of the new fantasized likelihood - if we call that new likelihood with another `observations_noise` kwarg, then this could be confusing. Not sure if that's a valid concern.